spec/mongo/query_cache_spec.rb

```diff
@@ -0,0 +1,313 @@
+require 'spec_helper'
+
+describe Mongo::QueryCache do
+
+  around do |spec|
+    Mongo::QueryCache.clear_cache
+    Mongo::QueryCache.cache { spec.run }
+  end
+
+  before do
+    authorized_collection.delete_many
+  end
+
+  let(:subscriber) { EventSubscriber.new }
+
+  let(:client) do
+    authorized_client.tap do |client|
+      client.subscribe(Mongo::Monitoring::COMMAND, subscriber)
+    end
+  end
+
+  let(:authorized_collection) { client['collection_spec'] }
+
+  describe '#enabled' do
+
+    context 'when query cache is disabled' do
+
+      before do
+        Mongo::QueryCache.enabled = false
+      end
+
+      it 'query cache is disabled' do
+        expect(Mongo::QueryCache.enabled?).to be(false)
+      end
+    end
+  end
+
+  describe '#cache' do
+
+    context 'when a block is uncached' do
+
+      before do
+        authorized_collection.insert_one({ name: 'testing' })
+      end
+
+      it 'block is executed with disabled query cache' do
+        Mongo::QueryCache.cache {
+          authorized_collection.find(name: 'testing')
+          expect(Mongo::QueryCache.enabled?).to be(true)
+        }
```

Comment:
It might also be good to set `Mongo::QueryCache.enabled = false` before the start of this test and then check that enabled is false again after the block is done executing.